### ISSUE TITLE
Use of jshint with SBT 1.3.0-RC1 fails to build.

### ISSUE THREAD

**Issue by gavinbaumanis** [sbt/sbt-jshint#44]
Excluding jshint from the build allows it complete successfully.
Using SBT 1.2.8, allows for a successful build, too.

Please find below the error;
```
>sbt update
OpenJDK 64-Bit Server VM warning: Ignoring option MaxPermSize; support was removed in 8.0
OpenJDK 64-Bit Server VM warning: Ignoring option MaxPermSize; support was removed in 8.0
[info] Loading global plugins from C:\Users\Gavin Baumanis\.sbt\1.0\plugins
[info] Loading settings for project pas-build from plugins.sbt ...
[info] Loading project definition from D:\workspace\ecpr\pas\project
[warn]
[warn]  Note: Unresolved dependencies path:
[error] sbt.librarymanagement.ResolveException: Error downloading org.webjars.npm:dom-serializer:[0,1)
[error]   Not found
[error]   not found: https://repo1.maven.org/maven2/org/webjars/npm/dom-serializer/%5B0,1)/dom-serializer-%5B0,1).pom
[error]   not found: C:\Users\Gavin Baumanis\.ivy2\local\org.webjars.npm\dom-serializer\[0,1)\ivys\ivy.xml
[error]   download error: Caught java.net.URISyntaxException: Illegal character in path at index 87: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml (Illegal character in path at index 87: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml
[error]         at lmcoursier.CoursierDependencyResolution.unresolvedWarningOrThrow(CoursierDependencyResolution.scala:227)
[error]         at lmcoursier.CoursierDependencyResolution.$anonfun$update$30(CoursierDependencyResolution.scala:196)
[error]         at scala.util.Either$LeftProjection.map(Either.scala:573)
[error]         at lmcoursier.CoursierDependencyResolution.update(CoursierDependencyResolution.scala:196)
[error]         at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:56)
[error]         at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:87)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:137)
[error]         at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:69)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$20(LibraryManagement.scala:150)
[error]         at scala.util.control.Exception$Catch.apply(Exception.scala:228)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:150)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:131)
[error]         at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:150)
[error]         at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:164)
[error]         at sbt.Classpaths$.$anonfun$updateTask0$5(Defaults.scala:2844)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:270)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error]         at sbt.Execute.work(Execute.scala:279)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:270)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error]         at java.base/java.lang.Thread.run(Thread.java:835)
[error] (update) sbt.librarymanagement.ResolveException: Error downloading org.webjars.npm:dom-serializer:[0,1)
[error]   Not found
[error]   not found: https://repo1.maven.org/maven2/org/webjars/npm/dom-serializer/%5B0,1)/dom-serializer-%5B0,1).pom
[error]   not found: C:\Users\Gavin Baumanis\.ivy2\local\org.webjars.npm\dom-serializer\[0,1)\ivys\ivy.xml
[error]   download error: Caught java.net.URISyntaxException: Illegal character in path at index 87: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml (Illegal character in path at index 87: https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/dom-serializer/[0,1)/ivys/ivy.xml
```

**Comment by marcospereira** [comment]
This can be https://github.com/coursier/coursier/issues/569.

What happens if you try with:

```scala
ThisBuild / useCoursier := false
```

?

**Comment by gavinbaumanis** [comment]
Hi there, 
Adding that to the top of my build.sbt - makes no difference.
I get exactly the same error thrown, sorry.

**Comment by alexarchambault** [comment]
Some related things were fixed in coursier by https://github.com/coursier/coursier/commit/3ae005a91033a52e30b1e1bb2673a6fa1a606d3d. So I guess this should be fixed with the next SBT 1.3.0 RC or release.

**Comment by ches** [comment]
Current Coursier (at the moment I'm touching a project using sbt-coursier 2.0.0-RC4, still on sbt 1.28) handles the version range syntax with a `[0, 1)`, but sbt-jshint fails to resolve because its npm dependencies conflict with each other:

```
[error] (coursierResolutions) coursier.error.ResolutionError$ConflictingDependencies: Conflicting dependencies:
[error] org.webjars.npm:domelementtype:[2.0.1,3) or [1,2) wanted by
[error]
[error]   org.webjars.npm:dom-serializer:0.2.1 wants [2.0.1,3)
[error]   └─ org.webjars.npm:domutils:1.5.1 wants org.webjars.npm:dom-serializer:[0,1)
[error]      └─ org.webjars.npm:htmlparser2:3.8.3 wants org.webjars.npm:domutils:[1.5,1.6)
[error]         └─ org.webjars.npm:jshint:2.9.3 wants org.webjars.npm:htmlparser2:[3.8,3.9)
[error]            └─ com.typesafe.sbt:sbt-jshint;sbtVersion=1.0;scalaVersion=2.12:1.0.6
[error]
[error]   org.webjars.npm:domhandler:2.3.0 wants [1,2)
[error]   └─ org.webjars.npm:htmlparser2:3.8.3 wants org.webjars.npm:domhandler:[2.3,2.4)
[error]      └─ org.webjars.npm:jshint:2.9.3 wants org.webjars.npm:htmlparser2:[3.8,3.9)
[error]         └─ com.typesafe.sbt:sbt-jshint;sbtVersion=1.0;scalaVersion=2.12:1.0.6
[error]
[error]   org.webjars.npm:domutils:1.5.1 wants [1,2)
[error]   └─ org.webjars.npm:htmlparser2:3.8.3 wants org.webjars.npm:domutils:[1.5,1.6)
[error]      └─ org.webjars.npm:jshint:2.9.3 wants org.webjars.npm:htmlparser2:[3.8,3.9)
[error]         └─ com.typesafe.sbt:sbt-jshint;sbtVersion=1.0;scalaVersion=2.12:1.0.6
[error]
[error]   org.webjars.npm:htmlparser2:3.8.3 wants [1,2)
[error]   └─ org.webjars.npm:jshint:2.9.3 wants org.webjars.npm:htmlparser2:[3.8,3.9)
[error]      └─ com.typesafe.sbt:sbt-jshint;sbtVersion=1.0;scalaVersion=2.12:1.0.6
[error]
[error]
[error] org.webjars.npm:entities:[2.0.0,3) or [1.0,1.1) wanted by
[error]
[error]   org.webjars.npm:dom-serializer:0.2.1 wants [2.0.0,3)
[error]   └─ org.webjars.npm:domutils:1.5.1 wants org.webjars.npm:dom-serializer:[0,1)
[error]      └─ org.webjars.npm:htmlparser2:3.8.3 wants org.webjars.npm:domutils:[1.5,1.6)
[error]         └─ org.webjars.npm:jshint:2.9.3 wants org.webjars.npm:htmlparser2:[3.8,3.9)
[error]            └─ com.typesafe.sbt:sbt-jshint;sbtVersion=1.0;scalaVersion=2.12:1.0.6
[error]
[error]   org.webjars.npm:htmlparser2:3.8.3 wants [1.0,1.1)
[error]   └─ org.webjars.npm:jshint:2.9.3 wants org.webjars.npm:htmlparser2:[3.8,3.9)
[error]      └─ com.typesafe.sbt:sbt-jshint;sbtVersion=1.0;scalaVersion=2.12:1.0.6
```

Indeed, trying to build sbt-jshint current master with sbt-coursier 2.0.0-RC4 enabled fails in the same way. I'll have a PR here in a moment.

**Comment by mkurz** [comment]
I just released [version 2.0.0](https://github.com/sbt/sbt-jshint/releases/tag/2.0.0) which uses latest sbt 1.9.7 which is using coursier by default. Everything is working fine, the scripted test passes. Seems coursier fixed their problems meanwhile.
So this pull request is not needed anymore IMHO.
